include/mbedtls/ssl_internal.h

```diff
@@ -387,8 +387,12 @@ unsigned char mbedtls_ssl_sig_from_pk( mbedtls_pk_context *pk );
 mbedtls_pk_type_t mbedtls_ssl_pk_alg_from_sig( unsigned char sig );
 #endif
 
-mbedtls_md_type_t mbedtls_ssl_md_alg_from_hash( unsigned char hash );
-unsigned char mbedtls_ssl_hash_from_md_alg( int md );
+mbedtls_md_type_t mbedtls_ssl_md_decode( unsigned char hash );
```

Comment:
Again, renaming a public function: even if it's defined in a header with "internal" in its name, so arguably not public, it's still visible to the linker, and I'm really not sure how it would affect people who use mbed TLS as a shared library.